agg-ast/schema_derivation/src/schema_derivation.rs

```diff
@@ -433,6 +434,61 @@ impl DeriveSchema for Stage {
             }
         }
 
+        fn rank_fusion_derive_schema(
+            rank_fusion: &RankFusion,
+            state: &mut ResultSetState,
+        ) -> Result<Schema> {
+            // 1. Derive the schema for each pipeline and union them together
+            let mut unioned_schema_pipelines: Schema = rank_fusion
+                .input
+                .pipelines
+                .values()
+                .map(|pipeline| {
+                    let derived_schema =
+                        derive_schema_for_pipeline(pipeline.clone(), None, &mut state.clone());
+                    derived_schema.unwrap_or(EMPTY_DOCUMENT.clone())
```

Comment:
If we encounter a schema derivation error, why are we ignoring it and using an `EMPTY_DOCUMENT` schema here? We may want to return an error if an error is encountered in any of the sub-pipelines.